Plugin/Model/ResourceModel/Catalog/GetValidCategoryPagesForSitemap.php

```diff
@@ -0,0 +1,259 @@
+        )->joinLeft(
+            ['url_rewrite' => $subject->getTable('url_rewrite')],
+            'e.entity_id = url_rewrite.entity_id AND url_rewrite.is_autogenerated = 1'
+            . $connection->quoteInto(' AND url_rewrite.store_id = ?', $store->getId())
+            . $connection->quoteInto(' AND url_rewrite.entity_type = ?', CategoryUrlRewriteGenerator::ENTITY_TYPE),
+            ['url' => 'request_path']
+        )->joinLeft(
+            ['x' => $showInSitemapAttributeDetails['table']],
+            'x.attribute_id = '. $showInSitemapAttributeDetails['attribute_id'] .' AND x.row_id = e.row_id',
+            ['xml' => 'x.value']
+        )->where(
+            'e.path LIKE ?',
+            $categoryRow['path'] . '/%'
+        )->where('x.value = ?', 1);
+
+        $this->addFilter($subject, $storeId, 'is_active', 1);
+
+        $query = $connection->query($this->_select);
+        while ($row = $query->fetch()) {
+            $category = $this->prepareCategory($subject, $row);
+            $categories[$category->getId()] = $category;
+        }
+
+        return $categories;
+    }
+
+    /**
+     * Get attribute details by attribute code
+     *
+     * @param string $attributeCode
+     * @return array
+     * @throws LocalizedException
+     */
+    public function getAttributeDetails(string $attributeCode): array
+    {
+        if (!isset($this->_attributesCache[$attributeCode])) {
+            $attribute = $this->categoryResource->getAttribute($attributeCode);
+
+            $this->_attributesCache[$attributeCode] = [
+                'entity_type_id' => $attribute->getEntityTypeId(),
+                'attribute_id' => $attribute->getId(),
+                'table' => $attribute->getBackend()->getTable(),
+                'is_global' => $attribute->getIsGlobal(),
+                'backend_type' => $attribute->getBackendType(),
+            ];
+        }
+        $attribute = $this->_attributesCache[$attributeCode];
+
+        return $attribute;
+    }
+
+    /**
+     * Prepare category
+     *
+     * Overide due to getCollection function required this function, also this function is protected visibility
+     *
+     * @param Category $subject
+     * @param array $categoryRow
+     * @return DataObject
+     * @throws LocalizedException
+     */
+    private function prepareCategory(Category $subject, array $categoryRow): DataObject
+    {
+        $category = new DataObject();
+        $category->setId($categoryRow[$subject->getIdFieldName()]);
+        $categoryUrl = !empty($categoryRow['url']) ? $categoryRow['url'] : 'catalog/category/view/id/' .
+            $category->getId();
+        $category->setUrl($categoryUrl);
+        $category->setUpdatedAt($categoryRow['updated_at']);
+        return $category;
+    }
+
+    /**
+     * Add attribute to filter
+     *
+     * Overide due to getCollection function required this function, also this function is protected visibility
```

Comment:
As above - override and protected vs. private

Reply:
Original core class contains  _addFilter method protected visibility. 

Original core method in `Magento\Sitemap\Model\ResourceModel\Catalog\Category`

`/**
     * Add attribute to filter
     *
     * @param int $storeId
     * @param string $attributeCode
     * @param mixed $value
     * @param string $type
     * @return \Magento\Framework\DB\Select|bool
     */
    protected function _addFilter($storeId, $attributeCode, $value, $type = '=')
    {
        $meta = $this->metadataPool->getMetadata(\Magento\Catalog\Api\Data\CategoryInterface::class);
        $linkField = $meta->getLinkField();

        if (!$this->_select instanceof \Magento\Framework\DB\Select) {
            return false;
        }

        if (!isset($this->_attributesCache[$attributeCode])) {
            $attribute = $this->_categoryResource->getAttribute($attributeCode);

            $this->_attributesCache[$attributeCode] = [
                'entity_type_id' => $attribute->getEntityTypeId(),
                'attribute_id' => $attribute->getId(),
                'table' => $attribute->getBackend()->getTable(),
                'is_global' => $attribute->getIsGlobal(),
                'backend_type' => $attribute->getBackendType(),
            ];
        }
        $attribute = $this->_attributesCache[$attributeCode];

        switch ($type) {
            case '=':
                $conditionRule = '=?';
                break;
            case 'in':
                $conditionRule = ' IN(?)';
                break;
            default:
                return false;
                break;
        }

        if ($attribute['backend_type'] == 'static') {
            $this->_select->where('e.' . $attributeCode . $conditionRule, $value);
        } else {
            $this->_select->join(
                ['t1_' . $attributeCode => $attribute['table']],
                'e.' . $linkField . ' = t1_' . $attributeCode . '.' . $linkField .
                ' AND t1_' . $attributeCode . '.store_id = 0',
                []
            )->where(
                't1_' . $attributeCode . '.attribute_id=?',
                $attribute['attribute_id']
            );

            if ($attribute['is_global']) {
                $this->_select->where('t1_' . $attributeCode . '.value' . $conditionRule, $value);
            } else {
                $ifCase = $this->getConnection()->getCheckSql(
                    't2_' . $attributeCode . '.value_id > 0',
                    't2_' . $attributeCode . '.value',
                    't1_' . $attributeCode . '.value'
                );
                $this->_select->joinLeft(
                    ['t2_' . $attributeCode => $attribute['table']],
                    $this->getConnection()->quoteInto(
                        't1_' .
                        $attributeCode .
                        '.' . $linkField . ' = t2_' .
                        $attributeCode .
                        '.' . $linkField . ' AND t1_' .
                        $attributeCode .
                        '.attribute_id = t2_' .
                        $attributeCode .
                        '.attribute_id AND t2_' .
                        $attributeCode .
                        '.store_id=?',
                        $storeId
                    ),
                    []
                )->where(
                    '(' . $ifCase . ')' . $conditionRule,
                    $value
                );
            }
        }

        return $this->_select;
    }`